tests/unit/test_group_project.py

```diff
@@ -230,6 +230,30 @@ def test_peer_grade_questions(self, expected_result, *stage_questions):
 
             get_stages.assert_called_with(PeerReviewStage.CATEGORY)
 
+    @ddt.data(
+        (None, []),
+        (None, [None, None, None]),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', None, '2016-10-29 04:30:48']),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', None, '2016-10-29 04:30:48', None]),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', '2016-10-29 04:30:48', '2016-02-29 04:30:48']),
+        ('2016-10-29 04:30:48', ['2016-10-29 04:30:48', '2016-02-26 04:30:48', '2016-02-29 04:30:48']),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', '2016-02-29 04:30:48', '2016-10-29 04:30:48']),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', '2016-10-29 04:30:48', '2016-10-29 04:30:48']),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', '2016-10-29 04:30:48', '2016-02-26 04:30:48']),
+        ('2016-10-29 04:30:48', ['2016-02-26 04:30:48', '2016-10-29 04:30:48', '2016-10-29 04:30:47']),
```

Comment:
@ziafazal sorry for being even more pedantic, but now it satisfies the following scenario:
1. If empty or all None - return None
2. Otherwise return '2016-10-29 04:30:48'

Some diversity in inputs and results would be nice to have.